### PR TITLE
Document procedure for old PR cleanup

### LINKED ISSUE #5634: Cleanup old pull requests?
Something that was in discussion during All Hands 2020 was what we should do with some of the old PRs within this repository.  There's a number of them that haven't been touched since around September.  We should discuss what we want to do with these PRs, as well as write up a policy in our Governance doc in regards to old PRs.

Pinging @chrisdavidmills, @Elchi3 and @ddbeck.

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
I figured I might as well attempt to document the procedure for cleaning up old pull requests!  This PR fixes #5634.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -61,6 +61,15 @@ Not everything is enforced or validated by the schema. A few things to pay atten
 - Nesting of feature identifiers should make sense.
 - Notes use correct grammar and spelling. They should be complete sentences ending with a period.
 
+### Old pull requests
+
+We may close pull requests that have been inactive for 30 or more days.  The procedure is as follows:
```
Comment (chrisdavidmills) [inline docs/contributing.md]: extra space — "days.  The" -> "days. The"

File: docs/contributing.md [hunk 2]
```diff
@@ -61,6 +61,15 @@ Not everything is enforced or validated by the schema. A few things to pay atten
...
+
+1. A peer will review the pull request, and potentially request changes or request more information.
```
Comment (chrisdavidmills) [inline docs/contributing.md]: repetition; remove second "request".

File: docs/contributing.md [hunk 3]
```diff
@@ -61,6 +61,15 @@ Not everything is enforced or validated by the schema. A few things to pay atten
...
+2. If the pull request author doesn't respond within a week, a peer may ping and ask if the author plans to return to the pull request.
+3. After another week or two, a peer may add a label to mark the pull request as inactive, as well as comment that it's been flagged for closing due to inactivity.
```
Comment (chrisdavidmills) [inline docs/contributing.md]: simplify wording - "inactive, as well as comment" -> "inactive and comment"

Comment (chrisdavidmills) [reply docs/contributing.md]: Also, is the actual label used simply "Inactive". May clarify by updating "add a label to mark" to "adding an "Inactive" label to mark"

File: docs/contributing.md [hunk 4]
```diff
@@ -61,6 +61,15 @@ Not everything is enforced or validated by the schema. A few things to pay atten
...
+4. After one final week, a peer may simply close the pull request.  Typically, the pull request would be closed after 30 days of inactivity.
```
Comment (chrisdavidmills) [inline docs/contributing.md]: I'd put the second sentence on a new line, outside the bullets; it is really a comment on the whole procedure, not part of this bullet.

File: docs/contributing.md (hunk 1 above)
Comment (chrisdavidmills) [inline docs/contributing.md]: Update "inactive for 30 or more days." to "inactive for an extended period of time." We mention the 30 day estimate in the last sentence; no point in saying it twice.

**Comment by vinyldarkscratch** [comment]
Thanks for the quick review!  I've now addressed

**Comment by Elchi3** [comment]
Does it make sense that responsibilities of owners and peers apply to the PR closing rule?
I saw you commented on script/infra PRs and plan to close them, but usually these would be for owners to make decisions on.

**Comment by vinyldarkscratch** [comment]
Yeah, you’ve got a point there — whoops!  I figured that those PRs I marked as inactive wouldn’t really be touched, but I’ll make sure to do it only to data changes from now on.

**Comment by vinyldarkscratch** [comment]
Actually, wait: I was just looking at the governance doc to double-check some things.  In the table of responsibilities, it states that peers can review pull requests and issues, but only owners can _merge_ infrastructure et. al. pull requests.  It doesn’t say that peers can’t review and close those pull requests.  Perhaps we need to start a discussion on that part?

**Comment by ddbeck** [comment]
Hey @vinyldarkscratch thanks for opening this PR. I think there's some good stuff here, but I've got some concerns.

But before I get into that, I think you have gotten ahead of the process a bit here. This PR hasn't been approved or merged yet, but it looks like you've [labeled several PRs as inactive](https://github.com/mdn/browser-compat-data/labels/inactive%20%F0%9F%92%A4) and made comments about closing some of them. I think this move is premature (though it has been illustrative about how this process might work). Speaking for myself, I'd appreciate it if you would remove the labels for now (and maybe link back to the discussion here, where you've commented).

OK, now to address the PR itself. I think this PR touches on more than one question. A few come to mind:

* What should we do when a PR author is unresponsive to requested changes?
* What should we do when a PR has gone un-reviewed for a long time?
* What should we do when a PR is blocked or stalled?

Superficially, these PRs all end up in the same place: an old PR. The proposed process here addresses the old PR symptom, but it doesn't really do anything about the multiple underlying causes.

I think we've informally established some practices for the first question: ask for follow up and if it's not coming, open a superseding PR or issue, thank the original author, and close the PR. It would be great to document that process, so we can all practice it the same way. I imagine we could come up with guidelines to help with the other questions as well.

But I'm uncomfortable with a mechanistic process for closing old PRs. For starters, I don't like the idea of closing PRs due to _maintainer_ inactivity. For example, infra PRs have historically taken much longer to respond to than data PRs; there are few suitable reviewers and priority, not age, tends to dominate the decision-making process. I worry that a mechanistic process could lead to infra PRs being closed without review before owners have had a chance to properly consider the proposal.

(Admittedly, the owners, myself included, could do a better job about making timely up-or-down decisions about long-standing proposals. I've already talked to @Elchi3 about carving out a little bit of time to start doing this.)

Overall, I think we might benefit more from a set of guidelines for reviewers that would help them un-stick a stuck PR than we might from a process that simply closes PRs faster.

**Comment by vinyldarkscratch** [comment]
Ack, you’re right @ddbeck, I did get a bit ahead of myself on this one.  I guess you could say that I got a little excited to go forward with this (and I’ve been doing this before, though not giving as much of a timeframe or taking advantage of a label).  I think I saw Chris’ approval and thought “alright, let’s start it up”!  (I also apologize for not responding until just now — I hadn’t gotten a notification, or I accidentally removed it.  The PRs that were labeled as inactive have all been closed and/or superseded already, but I will hold off on labeling anymore.)

**Comment by Elchi3** [comment]
I'm closing this PR. Daniel's questions are worth going after, but I don't think this PR addresses them. Let's discuss more in the issue.

Specifically, 

> What should we do when a PR author is unresponsive to requested changes?
> What should we do when a PR has gone un-reviewed for a long time?

are interesting questions to me and probably need two separate answers in our docs. I think the first question needs an answer that guides PR authors and the second question needs an answer that guides peers and owners. Both are currently lacking specific guidance, which *may* currently lead us to treat these situations differently which is not ideal. So, still a good idea to formalize some things but inactivity closure of PRs is not it.

**Pull request closed without merging**